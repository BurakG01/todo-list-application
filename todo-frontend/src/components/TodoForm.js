
import React from "react";
import Button from "react-bootstrap/Button";
import InputGroup from "react-bootstrap/InputGroup";
import FormControl from "react-bootstrap/FormControl";
import PropTypes from 'prop-types'
export default class TodoForm extends React.Component {

    static propTypes = {
        setTodo: PropTypes.func,
        createTodo: PropTypes.func.isRequired
    }

    render(){
        return (
            <InputGroup className="mb-3">
            <FormControl
              placeholder="add item . . . "
              size="lg"
              value={this.props.todo}
              onChange={({ target }) => this.props.setTodo(target.value)}
              aria-label="add something"
              aria-describedby="basic-addon2"
            />
            <InputGroup.Append>
              <Button variant="dark" size="lg" onClick={this.props.createTodo}>
                ADD
              </Button>
            </InputGroup.Append>
          </InputGroup>
        )
    }
}
