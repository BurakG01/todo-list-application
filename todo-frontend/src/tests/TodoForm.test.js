import TodoForm from "../components/TodoForm";


it("renders correctly", () => {
  const wrapper = shallow(
    <TodoForm setTodo={() => {}} createTodo={() => {}} />
  );

  expect(wrapper).toMatchSnapshot();
});

it("calls createTodo", () => {
  const spy = sinon.spy();

  const wrapper = mount(<TodoForm setTodo={() => {}} createTodo={spy} />);

  wrapper.find("Button").first().simulate("click");

  expect(spy.calledOnce).toBe(true);
});

it("calls setTodo", () => {
  const spy = sinon.spy();

  const wrapper = mount(<TodoForm setTodo={spy} createTodo={() => {}} />);

  wrapper.find("FormControl").first().simulate("change");

  expect(spy.calledOnce).toBe(true);
});
