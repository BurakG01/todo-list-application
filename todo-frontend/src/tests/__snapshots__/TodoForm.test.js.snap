// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<InputGroup
  className="mb-3"
>
  <FormControl
    aria-describedby="basic-addon2"
    aria-label="add something"
    onChange={[Function]}
    placeholder="add item . . . "
    size="lg"
  />
  <InputGroupAppend>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      size="lg"
      variant="dark"
    >
      ADD
    </Button>
  </InputGroupAppend>
</InputGroup>
`;
